adam-core/src/main/scala/org/bdgenomics/adam/rdd/read/AlignmentRecordRDD.scala

```diff
@@ -85,6 +86,35 @@ sealed trait AlignmentRecordRDD extends AvroReadGroupGenomicRDD[AlignmentRecord,
   }
 
   /**
+   * Convert this set of reads into Coverage records
+   *
+   * @return Returns a CoverageRDD where all reads have been grouped and
+   * counted at all overlapping positions
+   */
+  def toCoverage: CoverageRDD = {
+    val coverage: RDD[(ReferenceRegion, Int)] =
+      rdd.rdd
+        .flatMap(r => {
+          val t: List[Long] = List.range(r.getStart, r.getEnd)
+          val m: List[(ReferenceRegion, Int)] = t.map(n => (ReferenceRegion(r.getContigName, n, n + 1), 1))
+          m
+        }).reduceByKey(_ + _).sortByKey()
```

Comment:
Currently, this is storing an individual record for each position in the genome. I would like to run length encode this using ReferenceRegion. What would be the most efficient way to group by ReferenceRegion and count to merge adjacent regions?